src/python/pants/backend/python/goals/lockfile.py

```diff
@@ -35,17 +43,63 @@
     WrappedGenerateLockfile,
 )
 from pants.core.util_rules.lockfile_metadata import calculate_invalidation_digest
-from pants.engine.fs import CreateDigest, Digest, DigestContents, FileContent, MergeDigests
+from pants.engine.fs import (
+    CreateDigest,
+    Digest,
+    DigestContents,
+    DigestEntries,
+    FileContent,
+    FileEntry,
+    GlobMatchErrorBehavior,
+    MergeDigests,
+    PathGlobs,
+)
 from pants.engine.internals.synthetic_targets import SyntheticAddressMaps, SyntheticTargetsRequest
 from pants.engine.internals.target_adaptor import TargetAdaptor
 from pants.engine.process import ProcessCacheScope, ProcessResult
 from pants.engine.rules import Get, collect_rules, rule
 from pants.engine.target import AllTargets
 from pants.engine.unions import UnionRule
+from pants.option.option_types import BoolOption, StrListOption
+from pants.option.subsystem import Subsystem
 from pants.util.docutil import bin_name
 from pants.util.logging import LogLevel
 from pants.util.ordered_set import FrozenOrderedSet
 from pants.util.pip_requirement import PipRequirement
+from pants.util.strutil import softwrap
+
+logger = logging.getLogger(__name__)
+
+
+class PexLockSubsystem(Subsystem):
+    options_scope = "pex-lock"
+    help = "Pex lock specific Python lockfile options."
+
+    update = BoolOption(
```

Comment:
What if we used a different option here:
```
[pex-cli]
regenerate = false # default is true
```

This way, regenerate forces it to use the new lockfile codepath, otherwise it does a minimal lockfile update based on the updated requirements.

Though, I can also see the benefit of `pants generate-lockfiles --pex-lock-update --resolve=st2 --pex-lock-project=foobar` for surgical updates of the locked requirements without updating any of the requirements or constraints. Hmm. So I guess this PR implements the surgical lock update feature, not a minimal lockfile update based on modified requirements/constraints.

Does the pants metadata in the lockfile include info about the additional ad-hoc constraints used to surgically update the lockfile? Should it?